### PR TITLE
[feature] Visual Studio 2022 (ARM64EC, ARM64, x64)

### PR DESCRIPTION
**Pull request by danoli3**
Visual Studio 2022
Targeted against Latest dependency Libraries with all targets 

### Targets:
- x64
- ARM64

Experimental Target
- ARM64EC

Updated openFrameworks Project for Visual Studio 2022 - v143
Updated Template Project for Visual Studio 2022 - v143

ProjectGenerator Updated https://github.com/openframeworks/projectGenerator/pull/474
- Project Generator successful works for multitarget addons with all 3 targets folders correctly 

WinSDK 11 for ARM64EC
WinSDK 10 for others

### CODE REVIEW

**Comment by danoli3** [comment]
Okay this is all ready for go

Visual Studio 2022 
- x64 
- ARM64
- ARM64EC 

Runtime Tested - x64 / VS arm64
Debug / Release 
- Addons 
- Core Templates 
- Custom Projects

Likely some runtime issues with ARM64EC to figure out after merge

**Comment by danoli3** [comment]
Also GLM 9.9.9.9 define fix (2024 release)

File: addons/ofxAssimpModelLoader/src/ofxAssimpMeshHelper.h
Comment (danoli3) [inline addons/ofxAssimpModelLoader/src/ofxAssimpMeshHelper.h:1]: GLM constants fix keeping with define inline as recent changes on master for now

File: libs/openFrameworks/utils/ofConstants.h [hunk 1]
```diff
@@ -389,6 +389,12 @@ typedef TESSindex ofIndexType;
 	#endif
 #endif
 
+#if (defined(_M_ARM64) || defined(_M_ARM64EC)) && defined(TARGET_WIN32)
+	#undef USE_FMOD // No FMOD lib for ARM64 yet
+	#define TARGET_NO_SOUND
```
Comment (danoli3) [inline libs/openFrameworks/utils/ofConstants.h:394]: Fmod not supported on ARM64/ARM64EC currently

File: scripts/templates/vs/emptyExample.vcxproj [hunk 2]
```diff
@@ -69,18 +89,40 @@
...
     </ClCompile>
+    <PostBuildEvent>
+    </PostBuildEvent>
+  </ItemDefinitionGroup>
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
     <Link>
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <SubSystem>Console</SubSystem>
-      <RandomizedBaseAddress>false</RandomizedBaseAddress>
+      <RandomizedBaseAddress>true</RandomizedBaseAddress>
```
Comment (danoli3) [inline scripts/templates/vs/emptyExample.vcxproj:99]: Randomised Base required for ARM on MS

**Review by danoli3 — COMMENTED** [review]
Looking good, reviewed

File: libs/openFrameworks/utils/ofConstants.h (hunk 1 above)
Comment (ofTheo) [reply libs/openFrameworks/utils/ofConstants.h:394]: We have a Windows Media Foundation sound player now. 
So you could skip `#define TARGET_NO_SOUND` and it should work.

**Comment by ofTheo** [comment]
@danoli3 - this looks good to me too except for the sound comment I added above. 

Thanks!! :)

File: libs/openFrameworks/utils/ofConstants.h (hunk 1 above)
Comment (danoli3) [reply libs/openFrameworks/utils/ofConstants.h]: Oh yes! I just tested on ARM64 and yes that works with change to  #define OF_NO_FMOD which activates now into that media foundation! Awesome!

**Pull request merged**